src/System.Security.Cryptography.Encoding/src/Internal/Cryptography/OpenSslAsnFormatter.cs

```diff
@@ -51,12 +51,17 @@ protected override string FormatNative(Oid oid, byte[] rawData, bool multiLine)
                     int printLen = Interop.Crypto.GetMemoryBioSize(bio);
 
                     // Account for the null terminator that it'll want to write.
-                    StringBuilder builder = new StringBuilder(printLen + 1);
-                    Interop.Crypto.BioGets(bio, builder, builder.Capacity);
+                    var buf = new byte[printLen + 1];
+                    int read = Interop.Crypto.BioGets(bio, buf, buf.Length);
 
-                    return builder.ToString();
+                    if (read < 0)
+                    {
+                        throw Interop.Crypto.CreateOpenSslCryptographicException();
+                    }
```

Comment:
Note that error checking wasn't previously being done here.  I've added it, as I couldn't see anything that would prevent an error.  Let me know if this should be removed.

(Also, the docs seem to suggest that read == 0 would also be an error, but the other place we're using BioGets we're only checking for < 0, so I copied that here.  Let me know if both should be changed.)